lib/spack/spack/compilers/arm.py

```diff
@@ -35,7 +36,21 @@ class Arm(spack.compiler.Compiler):
     # InstalledDir:
     # /opt/arm/arm-hpc-compiler-19.0_Generic-AArch64_RHEL-7_aarch64-linux/bin
     version_argument = '--version'
-    version_regex = r'Arm C\/C\+\+\/Fortran Compiler version ([^ )]+)'
+    version_regex = r'Arm C\/C\+\+\/Fortran Compiler version ([\d\.]+) '\
+                    r'\(build number (\d+)\) '
+
+    @classmethod
+    def extract_version_from_output(cls, output):
+        """Extracts the version from compiler's output."""
+        match = re.search(cls.version_regex, output)
+        if match:
+            if match.group(1).count('.') == 1:
+                temp = match.group(1) + ".0." + match.group(2)
+            else:
+                temp = match.group(1) + "." + match.group(2)
+        else:
```

Comment:
Is there actually a situation where this else clause will be hit?

Reply:
The `else` clause can be avoided completely if we write things like:
```python
match = re.search(cls.version_regex, output)
temp = 'unknown'
if match:
    ...

return temp
```

Reply:
I can't think of a case where the compiler is found but there is not version number. 

I can re-write the compiler version function to eliminate the else clause as @alalazo suggests, but its functionality will be the same. None of the compiler version tests are written with a situation where "unknown" is returned. 

Please advise if you would like changes to the code. Thanks!

Reply:
Yes, please make the change. I think we can get your changes to 100% test coverage, which will help me sleep at night 😄

Reply:
I changed the compiler version function to @alalazo 's suggestion. There is no 'else' clause for the 'if match' check. Thanks for all the feedback!